From: Mark Nelson <mdnelson8@gmail.com>
Date: Sun, 22 Nov 2009 12:07:41 +1100
Subject: [PATCH] ahci: let users know that Promise PDC42819 support is limited to SATA devices
Git-commit: 7a02267e0f3938a6e91c0a0a82211e4d7007210e
References: FATE#311054
Patch-Mainline: 2.6.33

ahci can drive the Promise PDC42819, but obviously it can only use SATA
disks connected to this controller. The controller can actually support
SAS disks as well, but we only know how to use it in it's AHCI mode.

Add a message to let users know that because ahci is driving their chip
they can only use the SATA disks connected to this controller.

Signed-off-by: Mark Nelson <mdnelson8@gmail.com>
Signed-off-by: Jeff Garzik <jgarzik@redhat.com>
Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 drivers/ata/ahci.c |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/drivers/ata/ahci.c b/drivers/ata/ahci.c
index e3a5609..e3e5734 100644
--- a/drivers/ata/ahci.c
+++ b/drivers/ata/ahci.c
@@ -3083,6 +3083,14 @@ static int ahci_init_one(struct pci_dev *pdev, const struct pci_device_id *ent)
 	    pdev->subsystem_device == 0xcb89)
 		return -ENODEV;
 
+	/* Promise's PDC42819 is a SAS/SATA controller that has an AHCI mode.
+	 * At the moment, we can only use the AHCI mode. Let the users know
+	 * that for SAS drives they're out of luck.
+	 */
+	if (pdev->vendor == PCI_VENDOR_ID_PROMISE)
+		dev_printk(KERN_INFO, &pdev->dev, "PDC42819 "
+			   "can only drive SATA devices with this driver\n");
+
 	/* acquire resources */
 	rc = pcim_enable_device(pdev);
 	if (rc)
-- 
1.6.0.2

